test/files/run/repl-class-based-outer-pointers.scala

```diff
@@ -0,0 +1,30 @@
+import scala.tools.nsc.Settings
+import scala.tools.partest.ReplTest
+
+object Test extends ReplTest {
+  override def transformSettings(s: Settings) = {
+    s.Yreplclassbased.value = true
+    s
+  }
+
+  def code = """
+    |sealed abstract class Value; object Value {
+    |  final case class Num(value: Double) extends Value
+    |  final case class Str(value: String) extends Value
+    |  final case class Bool(value: Boolean) extends Value
+    |}
```

Comment:
Add another line `class C { final case class Num(value: Double) }` showing that in this case the warning is still there?